### PR TITLE
raftstore: fix apply check

### PR DESCRIPTION
**Pull request by BusyJay**
When the snapshot can be canceled directly, the peer_storage's snap_state should be updated.

@siddontang @zhangjinpeng1987 @hhkbp2 PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
Any test?

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -335,7 +335,13 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         for region_id in self.region_ranges.values() {
             let region = self.region_peers[region_id].region();
             let start_key = keys::enc_start_key(region);
-            try!(delete_all_in_range(&self.engine, &last_start_key, &start_key));
+            if last_start_key < start_key {
+                info!("[store {}] cleanup [{}, {})",
+                      self.store_id(),
+                      escape(&last_start_key),
+                      escape(&start_key));
+                try!(delete_all_in_range(&self.engine, &last_start_key, &start_key));
```
Comment (ngaut) [inline src/raftstore/store/store.rs]: We are cleaning the garbage data here, but i didn't see any explanation of garbage data.

Comment (BusyJay) [inline src/raftstore/store/store.rs]: This change is unrelated with this pr, I have removed it.

**Comment by BusyJay** [comment]
Unit test has been updated in [peer_storage.rs](https://github.com/pingcap/tikv/pull/1247/files#diff-ac5fa05712c478a85585b4852858d036R1260). The integrated test can't control the apply process precise enough, which makes it impossible to write a stable case.

File: src/raftstore/store/peer_storage.rs [hunk 2]
```diff
@@ -608,8 +608,7 @@ impl PeerStorage {
     #[inline]
     pub fn is_applying(&self) -> bool {
         match *self.snap_state.borrow() {
-            SnapState::Applying(_) |
-            SnapState::ApplyAborted => true,
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/peer_storage.rs]: Why `ApplyAborted` consider as applying in before, for any other considerations?

**Comment by siddontang** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
rest LGTM

**Comment by zhangjinpeng1987** [comment]
LGTM
